### PR TITLE
Ignore imports in ajava parsing

### PR DESCRIPTION
**Pull request by zcai1**
We currently require the imports in `*.ajava` to exactly match with the ones in the corresponding `*.java` source files. This requirement is error-prone and not very useful for providing any information, so this PR removes the requirement.

Fixed https://github.com/eisop/checker-framework/issues/360

### CODE REVIEW

**Comment by zcai1** [comment]
```
/__w/1/s/framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:3150: warning: use of default constructor, which does not provide a comment
```

Not sure why this only appears in misc_jdk18.

**Comment by wmdietl** [comment]
Thanks! Could you also open an issue that contains the stack trace from this error.
The work around in #356 is https://github.com/eisop/checker-framework/pull/356/commits/8d213ea5481205c9f666e2a359386a5b03f8c76e
Once #356 is merged, please pull in master into this PR.

File: build.gradle [hunk 1]
```diff
@@ -222,15 +222,25 @@ allprojects {
...
+                    'framework',
+                    'framework-test',
+                    'javacutil',
+            ]
+            targets = targets.collectMany {[
+                    // must call toString() to convert GString to String
+                    "${it}/**/*.java".toString(),
+                    "${it}/**/*.ajava".toString()
+            ]}
+            target targets
```
Comment (wmdietl) [inline build.gradle:242]: Should the `// TODO: Formatting of this` a few lines down be removed now?

File: framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java [hunk 2]
```diff
@@ -3148,6 +3148,15 @@ private void stubDebug(String warning) {
      * #processField}.
      */
     private class AjavaAnnotationCollectorVisitor extends DefaultJointVisitor {
+        @Override
+        public Void visitCompilationUnit(CompilationUnitTree javacTree, Node javaParserNode) {
+            CompilationUnit node = castNode(CompilationUnit.class, javaParserNode, javacTree);
+            processCompilationUnit(javacTree, node);
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:3154]: Please add a comment what this override is doing - or not doing - compared to the super method.

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for the improvement!

**Comment by wmdietl** [comment]
> ```
> /__w/1/s/framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java:3150: warning: use of default constructor, which does not provide a comment
> ```
> 
> Not sure why this only appears in misc_jdk18.

Just make the default constructor explicit and add some "Default constructor." javadoc. I think Java 18 is doing this as a step towards value classes.

File: build.gradle (hunk 1 above)
Comment (zcai1) [reply build.gradle:242]: Yes, it's removed now.

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -9,12 +9,15 @@ When reporting issues on an artificial tree (generated by the compiler), always
 try to find the closest non-artificial parent in the AST path to provide position
 information.
 
+Made formatting rules for *.ajava files to be consistent with the ones for *.java
+files. Ignored imports when parsing ajava files.
```
Comment (wmdietl) [inline docs/CHANGELOG.md:13]:
```suggestion
Formatting rules for `*.ajava` files are now consistent with the ones for `*.java` files.
Imports are now ignored when parsing `ajava` files.
```

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**